### ISSUE TITLE
enh(skore): Add a `decision_threshold` parameter to the thresholded classification metric

### ISSUE THREAD

**Issue by glemaitre** [probabl-ai/skore#2111]
### What would you like to say?

Currently, thresholded classification metrics like `accuracy` are using the implementation from scikit-learn.

Those methods, by default uses `model.predict` that is equivalent to `model.predict_proba(X) > 0.5` where 0.5 is the decision threshold. If one would like to change this behaviour, one needs to use the [`FixedThresholdClassifier`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.FixedThresholdClassifier.html) and change the decision threshold.

To avoid to request user to do this additional step, we could instead add a new parameter `decision_threshold` that would always call `model.predict_proba` instead and make the thresholding and then call the scikit-learn function.

In addition, this parameter could be exposed into the `report.metrics.summarize()` to compute bunch of metrics using this threshold and show it in the report of the display.

While this approach is certainly useful for binary classification, we need to put some thoughts a bit more when we tackle the multiclass classification to know if we should allow to pass such decision threshold in this case.
